File: 290-word-pattern/290-word-pattern.cpp
class Solution {
public:
    bool wordPattern(string pattern, string s) {
        unordered_map<char,string> mp1;
        vector<string> word;
        string res = "";
        for(int i=0;i<s.size();i++){
            if(s[i] != ' '){
                res += s[i];
            }else if(s[i] == ' '){
                word.push_back(res);
                res = "";
            }
        }
        word.push_back(res);
        if(pattern.length() != word.size()) return 0;
        for(int i = 0; i < word.size(); i++){
            if(mp1[pattern[i]] == ""){
                for(auto &j : mp1){
                    if(j.second == word[i]) return 0;
                }
                mp1[pattern[i]] = word[i];
            }else if(mp1[pattern[i]] != word[i]) return 0;
        }
        return 1;
    }
};

File: 290-word-pattern/290-word-pattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        dic = defaultdict()
        words = s.split(' ')
        if len(pattern) != len(words) or len(set(words)) != len(set(pattern)):
            return 0
        for i,c in enumerate(pattern):
            if c in dic:
                if dic[c] != words[i]:
                    return False
            else:
                dic[c] = words[i]
        return True
        